public/lesswrong-editor/src/ckeditor5-footnote/src/footnoteEditing/autoformatting.js

```diff
@@ -0,0 +1,81 @@
+// @ts-check
+import inlineAutoformatEditing from '@ckeditor/ckeditor5-autoformat/src/inlineautoformatediting';
+import { Editor } from '@ckeditor/ckeditor5-core';
+import Element from '@ckeditor/ckeditor5-engine/src/model/element';
+import Text from '@ckeditor/ckeditor5-engine/src/model/text';
+import TextProxy from '@ckeditor/ckeditor5-engine/src/model/textproxy';
+import Range from '@ckeditor/ckeditor5-engine/src/model/range';
+import { modelQueryElement, modelQueryElementsAll } from '../utils';
+
+export const addFootnoteAutoformatting = (editor, rootElement) => {
+	if(editor.plugins.has('Autoformat')) {
+		const autoformatPluginInstance = editor.plugins.get('Autoformat');
+		inlineAutoformatEditing(editor, autoformatPluginInstance, 
+			matchingCallback, 
+			(_, /**@type Range[]*/ ranges) => formatCallback(ranges, editor, rootElement)
+			);
+	}
+}
+
+/**
+ * @param {string} text 
+ * @returns {{remove: number[][], format: number[][]}}
+ */
+const matchingCallback = (text) => {
+	const results = text.match(/\[\^([0-9]+)\]/);
```

Comment:
I feel nervous about this. I agree it's somewhat rare to have text like `[12]` without it being a footnote, but I would feel more comfortable if we matched based on a `data-` HTML attribute or something similar.

Reply:
Ben and I talked about this in person, but for others: data attributes aren't an option here, but it's also a pretty specific case: this is matching text as it's being typed, and it only triggers if the string of the form e.g. `"[^1]"`, where the number has to be a valid existing footnote index or the next unused index.